akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/ActorTestKit.scala

```diff
@@ -131,19 +155,19 @@ object ActorTestKit {
  *
  * For synchronous testing of a `Behavior` see [[BehaviorTestKit]]
  */
-final class ActorTestKit private[akka] (val name: String, val config: Config, settings: Option[TestKitSettings]) {
+final class ActorTestKit private[akka] (val internalSystem: ActorSystem[_], settings: Option[TestKitSettings]) {
 
   // avoid slf4j noise by touching it first from single thread #28673
-  LoggerFactory.getLogger(name).debug("Starting ActorTestKit")
+  LoggerFactory.getLogger(internalSystem.name).debug("Starting ActorTestKit")
 
   implicit def testKitSettings: TestKitSettings =
     settings.getOrElse(TestKitSettings(system))
 
   /**
    * INTERNAL API
    */
-  @InternalApi private[akka] val internalSystem: ActorSystem[ActorTestKitGuardian.TestKitCommand] =
-    ActorSystem(ActorTestKitGuardian.testKitGuardian, name, config)
+  @InternalApi private[akka] val internalTestKitGuardian: ActorRef[ActorTestKitGuardian.TestKitCommand] =
+    internalSystem.systemActorOf(ActorTestKitGuardian.testKitGuardian, "test")
```

Comment:
This isn't used as a "Guardian" any more. It is always spawned as a system actor so we should rename this. We can do it in this PR or we can follow up later.